### PR TITLE
External wallet optiomizations

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/10536

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
- make sure that uphold flow still works (connecting to uphold, transfer funds, doing a tip)

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 1]
```diff
@@ -3151,14 +3151,18 @@ void RewardsServiceImpl::SaveExternalWallet(const std::string& wallet_type,
   new_wallet.SetStringKey("one_time_string", wallet->one_time_string);
   new_wallet.SetStringKey("user_name", wallet->user_name);
   new_wallet.SetBoolKey("transferred", wallet->transferred);
+  new_wallet.SetStringKey("verify_url", wallet->verify_url);
+  new_wallet.SetStringKey("add_url", wallet->add_url);
+  new_wallet.SetStringKey("withdraw_url", wallet->withdraw_url);
+  new_wallet.SetStringKey("account_url", wallet->account_url);
```
Comment (kylehickinson) [inline components/brave_rewards/browser/rewards_service_impl.cc:3157]: I assume these have to be added to iOS as well? Current changes for iOS only include updating the client method

Comment (NejcZdovc) [reply components/brave_rewards/browser/rewards_service_impl.cc:3157]: correct we need to add it to iOS as well

Comment (kylehickinson) [reply components/brave_rewards/browser/rewards_service_impl.cc:3157]: Am in the middle of a 1.11.x build for rewards internals but can add it myself later. If you want to give it a go you will need to update the methods in `BATExternalWallet+DictionaryValue.mm`

Comment (NejcZdovc) [reply components/brave_rewards/browser/rewards_service_impl.cc:3157]: sure let me do it

**Comment by NejcZdovc** [comment]
CI failed on audit-deps and test-install, all of this known

**Pull request merged**